corporate/views/upgrade.py

```diff
@@ -153,6 +153,8 @@ def sponsorship(
 ) -> HttpResponse:
     realm = user.realm
     billing_session = RealmBillingSession(user)
+    # Ensures customer is created first before updating sponsorship status.
+    billing_session.update_customer_sponsorship_status(True)
```

Comment:
This should I think still be in the `form.is_valid` block -- as written I think this marks as pending after the GET request, without waiting for the user to submit the form.

Reply:
Yeah, fixed.